UPGRADE_GUIDE.md

```diff
@@ -2,7 +2,203 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-csharp` library between major versions.
 
--   [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
+- [Upgrading from 3.x to 4.x](#upgrading-from-3x-to-40)
+- [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
+
+## Upgrading from 3.x to 4.0
+
+### 4.0 High Impact Changes
+
+- [Updating Dependencies](#40-updating-dependencies)
+- [Client Instance](#40-client-instance)
+- [Language Conventions](#40-language-conventions)
+- [Nullable Properties](#40-nullable-properties)
+
+### 4.0 Medium Impact Changes
+
+- [Error Types](#40-error-types)
+- [Beta Client](#40-beta-client)
+
+### 4.0 Low Impact Changes
+
+- [Services and Resources](#40-services-and-resources)
+
+## 4.0 Updating Dependencies
+
+*Likelihood of Impact: **High***
+
+**Dependencies**
+
+- RestSharp was upgraded from v107 to v108
+- All dependencies had minor version bumps
```

Comment:
Thought: I'd maybe also callout that all deps are now locked to a major version.

Reply:
Resolved